sdk/storage/azure-storage-blob-cryptography/src/test/java/com/azure/storage/blob/specialized/cryptography/EncryptedBlobClientBuilderTest.groovy

```diff
@@ -56,6 +57,31 @@ class EncryptedBlobClientBuilderTest extends Specification {
             .verifyComplete()
     }
 
+    /**
+     * Tests that a user application id will be honored in the UA string when using the encrypted blob client builder's default
+     * pipeline.
+     */
+    def "Encrypted blob client custom application id in UA string"() {
+        when:
+        def randomData = new byte[256]
+        new SecureRandom().nextBytes(randomData)
+
+        def encryptedBlobClient = new EncryptedBlobClientBuilder()
+            .endpoint(endpoint)
+            .containerName("container")
+            .blobName("blob")
+            .credential(credentials)
+            .key(new FakeKey("keyId", randomData), "keyWrapAlgorithm")
+            .httpClient(new ApplicationIdUAStringTestClient())
+            .httpLogOptions(new HttpLogOptions().setApplicationId("custom-id"))
```

Comment:
We have discussed more about this in last 2 weeks with @JonathanGiles  and wanted more generic and common approach to provide applicationId. I would suggest not to implement this in HttpLogOptions because we are going to deprecate this.
Here is Epic related to this issue https://github.com/Azure/azure-sdk-for-java/issues/15107

We will be releasing `com.azure.core.util.ClientOptions` in master branch just after September release. Here is the PR for it  https://github.com/Azure/azure-sdk-for-java/pull/13697/files#diff-860f082a6593b9a09fa1a4d4fd0b5cf7R105 

I understand that we are changing this mid-way of this PR but want to make sure we are future proof.